src/Propel/Runtime/Map/TableMap.php

```diff
@@ -294,7 +295,8 @@ public function getClassName()
     public function getCollectionClassName()
     {
         $collectionClass = $this->getClassName() . 'Collection';
-        if (class_exists($collectionClass)) {
+        $parentClasses = class_parents($collectionClass);
+        if (class_exists($collectionClass) && in_array(Collection::class, $parentClasses)) {
```

Comment:
Just realized, you can use `is_subclass_of()` instead of `class_parent()` and `in_array()`. Should fix the failing tests, too

Reply:
Hey @nederdirk, did you abandon this? Everything ok?

Reply:
Hye @mringler, thanks for your concern. I'm doing most of this upstream OSS work in my spare time, of which I haven't had much between work, kids and the rest of life. ;)

Reply:
Happy to hear, busy is good.
Do you still want to finish this? If not, I can easily do it, shouldn't take me more than a couple of minutes

Reply:
Oh yeah, please do so; I don't know if you can commit to my branch, but you might open a new PR, just as I have done for the original patch in #1660

Reply:
Great! I am cursed with the urge to close off things I get involved with, so that helps. Damn OCD.

Did not use any of the existing patches, since they each contain the same one line, and it was not working.